### PR TITLE
Gcc6 fixes

### PR DESCRIPTION
**Pull request by Moiman**

### CODE REVIEW

File: media-sound/mac/files/mac-3.99.4.5.7-gcc6.patch [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+--- mac-3.99-u4-b5-s7/src/Shared/NoWindows.h.old	2016-09-18 21:37:56.049105791 -0400
++++ mac-3.99-u4-b5-s7/src/Shared/NoWindows.h	2016-09-18 21:55:29.666295506 -0400
+@@ -39,8 +39,8 @@
+ typedef const wchar_t *     LPCWSTR;
+ 
+ #define ZeroMemory(POINTER, BYTES) memset(POINTER, 0, BYTES);
+-#define max(a,b)    (((a) > (b)) ? (a) : (b))
+-#define min(a,b)    (((a) < (b)) ? (a) : (b))
++#define max_macro(a,b)    (((a) > (b)) ? (a) : (b))
++#define min_macro(a,b)    (((a) < (b)) ? (a) : (b))
```
Comment (SoapGentoo) [inline media-sound/mac/files/mac-3.99.4.5.7-gcc6.patch]: maybe just consider using `std::min` and `std::max` over macros, which generally suck, your choice though

File: media-sound/mac/mac-3.99.4.5.7-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+RESTRICT="mirror"
+
+src_prepare() {
+	default
+	sed -i -e 's:-O3::' configure || die
+}
+
+pkg_setup() {
+	append-cppflags -DSHNTOOL
+	use cpu_flags_x86_mmx && append-ldflags -Wl,-z,noexecstack
```
Comment (SoapGentoo) [inline media-sound/mac/mac-3.99.4.5.7-r2.ebuild]: do this in `src_configure` before econf and nix `pkg_setup`

File: media-sound/mac/mac-3.99.4.5.7-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		$(use_enable cpu_flags_x86_mmx assembly)
+}
+
+src_install() {
+	default
+
+	insinto /usr/include/${PN}
+	doins src/MACLib/{BitArray,UnBitArrayBase,Prepare}.h #409435
+
+	prune_libtool_files --all
```
Comment (SoapGentoo) [inline media-sound/mac/mac-3.99.4.5.7-r2.ebuild]: nix this and use `find` instead: https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Handling_Libtool_Archives You can likely then remove `eutils`

File: x11-libs/gtkmathview/gtkmathview-0.8.0-r3.ebuild [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		--enable-builder-cache \
+		--enable-breaks \
+		--enable-boxml \
+		--disable-gmetadom \
+		--disable-static
+}
+
+src_install() {
+	default
+	prune_libtool_files
```
Comment (SoapGentoo) [inline x11-libs/gtkmathview/gtkmathview-0.8.0-r3.ebuild]: same here

File: media-sound/mac/mac-3.99.4.5.7-r2.ebuild (hunk 3 above)
Comment (Moiman) [inline media-sound/mac/mac-3.99.4.5.7-r2.ebuild]: Good eutils isn't needed then anymore.

File: media-sound/mac/files/mac-3.99.4.5.7-gcc6.patch (hunk 1 above)
Comment (Moiman) [inline media-sound/mac/files/mac-3.99.4.5.7-gcc6.patch]: I did leave it.

**Comment by billie80** [comment]
I have bumped media-sound/mac to 4.11.4.5.7 including an adapted version of your gcc6 patch. It still makes sense to apply this patch to 3.99.4.5.7 as well.

File: media-sound/gmorgan/files/gmorgan-0.79-remove-old-docs.patch [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+ datadir = $(prefix)/share/${PACKAGE}/doc
+-data_DATA=gmorgan0.75.pdf \
+-gmorgan0.75.odt \
+-gmorgan-old-es.pdf \
+-gmorgan-old-es.sxw \
+-gmorgan-old-fr.pdf \
+-gmorgan-old-fr.sxw \
+-gmorgan-old.pdf \
+-gmorgan-old.sxw 
++data_DATA=gmorgan0.75.pdf
```
Comment (SoapGentoo) [inline media-sound/gmorgan/files/gmorgan-0.79-remove-old-docs.patch]: also, you sure this shouldn't be `dist_doc_DATA`? PDF files should usually be installed into `$(docdir)`

Comment (Moiman) [inline media-sound/gmorgan/files/gmorgan-0.79-remove-old-docs.patch]: You are right. Fixed now.

**Pull request closed without merging**